embedchain/vectordb/vector_db.py

```diff
@@ -0,0 +1,159 @@
+from typing import Any, List, Optional, Union
+
+from langchain.docstore.document import Document
+
+from embedchain.vectordb.chroma_db import ChromaDB
+from embedchain.vectordb.elasticsearch_db import EsDB
+
+
+class VectorDb:

```

Comment:
The idea to abstract the common database methods is very good, and something I asked for. However, instead of having methods for all databases in one class, why don't we use the child classes to implement these methods? 

What I mean is, right now you have 

```python
def get(self, db_type):
    if self.db_type == 'es':
        # Do this
        pass
    elif self.db_type == 'chroma':
        # Do that
        pass
```

Why don't you implement

```
class EsDb:
    def get(self):
        # do this
        pass
```

etc. so use inheritance.

Reply:
update: removed new class, each vector database now inherits and define get, add, query, count, reset functionality.